### PR TITLE
#2777 Converted UserAgentInterceptor.java to UserAgentInterceptor.kt

### LINKED ISSUE #2777: convert UserAgentInterceptor.java to Kotlin
Parent issue: #1826

**What changes you will do:**
Convert UserAgentInterceptor Java file to Kotlin

### PR DESCRIPTION
**Pull request by vivek-gite**
Fixes #2777

### CODE REVIEW

File: core/src/main/java/org/kiwix/kiwixmobile/core/data/remote/UserAgentInterceptor.kt [hunk 1]
```diff
@@ -15,28 +15,21 @@
...
-    this.useragent = useragent;
-  }
-
-  @Override
-  public Response intercept(Chain chain) throws IOException {
-    Request originalRequest = chain.request();
-    Request newUserAgent = originalRequest.newBuilder().header("User-Agent", useragent).build();
-    return chain.proceed(newUserAgent);
+class UserAgentInterceptor(val useragent: String) : Interceptor {
+  @kotlin.jvm.Throws(IOException::class)
```
Comment (gouri-panda) [inline core/src/main/java/org/kiwix/kiwixmobile/core/data/remote/UserAgentInterceptor.kt:29]: We can import this `@kotlin.jvm.`

File: core/src/main/java/org/kiwix/kiwixmobile/core/data/remote/UserAgentInterceptor.kt [hunk 2]
```diff
@@ -15,28 +15,21 @@
...
+  override fun intercept(chain: Interceptor.Chain): Response {
+    val originalRequest: Request = chain.request()
```
Comment (gouri-panda) [inline core/src/main/java/org/kiwix/kiwixmobile/core/data/remote/UserAgentInterceptor.kt:31]: We Don't need to specify type here.

File: core/src/main/java/org/kiwix/kiwixmobile/core/data/remote/UserAgentInterceptor.kt [hunk 3]
```diff
@@ -15,28 +15,21 @@
...
-  }
-
-  @Override
-  public Response intercept(Chain chain) throws IOException {
-    Request originalRequest = chain.request();
-    Request newUserAgent = originalRequest.newBuilder().header("User-Agent", useragent).build();
-    return chain.proceed(newUserAgent);
+class UserAgentInterceptor(val useragent: String) : Interceptor {
+  @kotlin.jvm.Throws(IOException::class)
+  override fun intercept(chain: Interceptor.Chain): Response {
```
Comment (gouri-panda) [inline core/src/main/java/org/kiwix/kiwixmobile/core/data/remote/UserAgentInterceptor.kt:30]: Here too `Interceptor`.

**Comment by kelson42** [comment]
@vivek-gite Do you plan to fix this PR?

**Comment by gouri-panda** [comment]
@vivek-gite Feel free to open this when you are free :)

**Pull request closed without merging**